File: C++/variables.cpp
#include <iostream>
using namespace std;

int main() {
    int a = 10, b = 4;
    cout << "La primera variable es: " << a << endl;
    cout << "El signo de la operación es: * " << endl;
    cout << "La segunda variable es: " << b << endl;
    int c = a * b;
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de dato: int" << endl; 

    cout << "La primera variable es: " << a << endl;
    cout << "El signo de la operación es: / " << endl;
    cout << "La segunda variable es: " << b << endl;
    double d = (double)a / b;
    cout << "El resultado es: " << d << endl;
    cout << "Tipo de dato: double" << endl; 

    cout << "La primera variable es: " << a << endl;
    cout << "El signo de la operación es: + " << endl;
    cout << "La segunda variable es: " << b << endl;
    c = a + b;
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de dato: int" << endl; 

    cout << "La primera variable es: " << a << endl;
    cout << "El signo de la operación es: - " << endl;
    cout << "La segunda variable es: " << b << endl;
    c = a - b;
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de dato: int" << endl; 

    cout << "La primera variable es: " << a << endl;
    cout << "El signo de la operación es: ** (potencia en C++ usa pow)" << endl;
    cout << "La segunda variable es: " << b << endl;
    double e = pow(a, b);
    cout << "El resultado es: " << e << endl;
    cout << "Tipo de dato: double" << endl; 

    cout << "La primera variable es: " << a << endl;
    cout << "El signo de la operación es: % " << endl;
    cout << "La segunda variable es: " << b << endl;
    c = a % b;
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de dato: int" << endl; 

    return 0;
}

File: rubi/variables.rb
# Declaración de variables
a = 10
b = 4

# Multiplicación
puts "La primera variable es: #{a}"
puts "El signo de la operación es: *"
puts "La segunda variable es: #{b}"
c = a * b
puts "El resultado es: #{c}"
puts "Tipo de dato: Integer"

# División
puts "\nLa primera variable es: #{a}"
puts "El signo de la operación es: /"
puts "La segunda variable es: #{b}"
d = a.to_f / b.to_f # Conversión a float para precisión
puts "El resultado es: #{d}"
puts "Tipo de dato: Float"

# Suma
puts "\nLa primera variable es: #{a}"
puts "El signo de la operación es: +"
puts "La segunda variable es: #{b}"
c = a + b
puts "El resultado es: #{c}"
puts "Tipo de dato: Integer"

# Resta
puts "\nLa primera variable es: #{a}"
puts "El signo de la operación es: -"
puts "La segunda variable es: #{b}"
c = a - b
puts "El resultado es: #{c}"
puts "Tipo de dato: Integer"

# Potencia
puts "\nLa primera variable es: #{a}"
puts "El signo de la operación es: **"
puts "La segunda variable es: #{b}"
e = a ** b # Potencia en Ruby
puts "El resultado es: #{e}"
puts "Tipo de dato: Float"

# Módulo
puts "\nLa primera variable es: #{a}"
puts "El signo de la operación es: %"
puts "La segunda variable es: #{b}"
c = a % b
puts "El resultado es: #{c}"
puts "Tipo de dato: Integer"